packages/core/src/__tests__/index.ts

```diff
@@ -54,80 +54,95 @@ describe('@reatom/core', () => {
-          const keys = Object.keys(state)
-          return keys.length === 1 && keys[0].includes(name)
-        })(),
-      ).toBe(true)
-      expect(declareAtom([name], initialState, () => {})()).toEqual({
-        [name]: initialState,
+    describe('declareAtom', () => {
+      test('basics', () => {
+        const name = '_atomName_'
+        const initialState = {}
+        const atom = declareAtom(name, initialState, () => {})
+        const state = atom({}, initAction)
+
+        expect(getState(state, atom)).toBe(initialState)
+        expect(
+          (() => {
+            const keys = Object.keys(state)
+            return keys.length === 1 && keys[0].includes(name)
+          })(),
+        ).toBe(true)
+        expect(declareAtom([name], initialState, () => {})()).toEqual({
+          [name]: initialState,
+        })
       })
-    })
-    test('declareAtom', () => {
-      const addUnderscore = declareAction()
-      const atom1 = declareAtom(['name1'], '1', reduce => [
-        reduce(addUnderscore, state => `_${state}`),
-      ])
-      const atom2 = declareAtom(['name2'], '2', reduce => [
-        reduce(addUnderscore, state => `_${state}`),
-      ])
-      const atomRoot = combine([atom1, atom2])
-
-      let state = atomRoot()
-      expect(state).toEqual({
-        name1: '1',
-        name2: '2',
-        [getTree(atomRoot).id]: ['1', '2'],
+      test('strict uid', () => {
+        const addUnderscore = declareAction()
+        const atom1 = declareAtom(['name1'], '1', reduce => [
+          reduce(addUnderscore, state => `_${state}`),
+        ])
+        const atom2 = declareAtom(['name2'], '2', reduce => [
+          reduce(addUnderscore, state => `_${state}`),
+        ])
+        const atomRoot = combine([atom1, atom2])
+
+        let state = atomRoot()
+        expect(state).toEqual({
+          name1: '1',
+          name2: '2',
+          [getTree(atomRoot).id]: ['1', '2'],
+        })
+
+        state = atomRoot(state, addUnderscore())
+        expect(state).toEqual({
+          name1: '_1',
+          name2: '_2',
+          [getTree(atomRoot).id]: ['_1', '_2'],
+        })
       })
+      test('throw error if declareAtom called with undefined initial state', () => {
+        const run = () => declareAtom(['test'], undefined, reduce => [])
 
-      state = atomRoot(state, addUnderscore())
-      expect(state).toEqual({
-        name1: '_1',
-        name2: '_2',
-        [getTree(atomRoot).id]: ['_1', '_2'],
+        expect(run).toThrowError(
+          `[reatom] Atom "test". Initial state can't be undefined`,
+        )
       })
-    })
-    test('throw error if declareAtom called with undefined initial state', () => {
-      const run = () => declareAtom(['test'], undefined, reduce => [])
+      test('throw error if atom produced undefined value', () => {
+        const action = declareAction()
+
+        expect(() =>
+          declareAtom(['myAtom'], {}, r => r(action, () => undefined as any))(
```

Comment:
The current file uses `reduce`, `r` and `on`. The desired uniformity so as not to drive you crazy readers.

Reply:
I suggest to fix `on`